WalletLibrary/WalletLibrary/Networking/Resolvers/OpenID4VCIPreAuthTokenResolver.swift

```diff
@@ -8,27 +8,33 @@
  */
 struct OpenID4VCIPreAuthTokenResolver
 {
+    
+    private struct Constants
+    {
+        static let GrantType = "urn:ietf:params:oauth:grant-type:pre-authorized_code"
+    }
+    
     private let configuration: LibraryConfiguration
     
     init(configuration: LibraryConfiguration) 
     {
         self.configuration = configuration
     }
     
-    func resolve(using credentialOfferGrant: CredentialOfferGrant) async throws -> String
+    func resolve(using credentialOfferGrant: CredentialOfferGrant,
+                 pin: String? = nil) async throws -> String
     {
-        let url = try URL.getRequiredProperty(property: URL(string: credentialOfferGrant.authorization_server),
-                                              propertyName: "authorization_server")
-        
         let code = try CredentialOfferGrant.getRequiredProperty(property: credentialOfferGrant.pre_authorized_code,
```

Comment:
Looking at this file, I wasn't able to find the part where we check that the authorization server in the credential offer is contained in the credential issuer metadata's authorization servers. This could very possibly be already happening elsewhere, so please let me know if im totally missing it 😄 

This does bring to another suggestion, though. I really liked how Daniel phrased the steps here:

![image](https://github.com/user-attachments/assets/8e7dd067-b152-4942-9e86-b634a4e2684c)

Would it be possible to add comments to the code to make it obvious for everyone (including authenticator newbies like me) where each step is happening?

(Btw: step 5 should say "Make sure authorization server from credential offer is contained in authorization servers from credential issuer metadata.")

Reply:
this check is in a completely different file: https://github.com/microsoft/entra-verifiedid-wallet-library-ios/blob/9df3d5f2dfb3086fad7bc12e7a36d4fc5a13bfa9/WalletLibrary/WalletLibrary/Requests/Processors/OpenId4VCIProcessor.swift#L60 

Already have those comments :)